### PR TITLE
Added flag in Status to wait for backend, and fixed panic issue.

### PR DESCRIPTION
**Pull request by pratikspatil024**
Added an optional flag (`-w`) to the `bor status` command, which waits for at max 10 seconds (5 iterations of 2 seconds each) if the backend is not ready.

`bor status -w` and `bor status` both of these commands are valid.

### CODE REVIEW

File: internal/cli/server/service.go [hunk 1]
```diff
@@ -165,6 +165,10 @@ func (s *Server) ChainSetHead(ctx context.Context, req *proto.ChainSetHeadReques
 }
 
 func (s *Server) Status(ctx context.Context, _ *empty.Empty) (*proto.StatusResponse, error) {
+	if s.backend == nil {
+		return nil, fmt.Errorf("the service is currently unavailable, try again later")
```
Comment (JekaMas) [inline internal/cli/server/service.go:169]: why fmt.Errorf ?

Comment (pratikspatil024) [reply internal/cli/server/service.go]: Changed to `errors.New()`, thanks.

Comment (JekaMas) [reply internal/cli/server/service.go]: Also possible to define a package level error variable and reuse it here.

Comment (pratikspatil024) [reply internal/cli/server/service.go]: Done, thanks!

File: internal/cli/server/service.go [hunk 2]
```diff
@@ -23,6 +24,8 @@ import (
 
 const chunkSize = 1024 * 1024 * 1024
 
+var ErrUnavailable = errors.New("the service is currently unavailable, try again later")
```
Comment (cffls) [inline internal/cli/server/service.go:27]: Let's be more specific on this error, e.g.

```
var ErrBorServiceUnavailable = errors.New("Bor service is currently unavailable, try again later")
```

Comment (pratikspatil024) [reply internal/cli/server/service.go]: Done, thanks!

**Review by manav2401 — APPROVED** [review]
LGTM. Do you think we should add a test for the status command?

**Comment by pratikspatil024** [comment]
Added a test to check if the command is successful or not. But this does not cover the whole code, as the backend gets available before we can get the grpc port. A solution for this is to break the `CreateMockServer` into 2 parts, the first one will allocate the ports, and the second one will use those ports, and in between, we can make a call to the Status command.

**Review by cffls — APPROVED** [review]

**Pull request merged**